Dockerfile

```diff
@@ -1,29 +1,37 @@
-FROM python:3.8
+# Stage 1: Start from the official Python-3.8 image: https://hub.docker.com/_/python/
+FROM python:3.8 AS base
```

Comment:
Love the addition of multistage Docker builds. This will help us out with debugging in case a step goes wrong.